src/frontend/src/handler/alter_table_column.rs

```diff
@@ -171,8 +177,11 @@ pub async fn handle_alter_table_column(
                 ))?
             }
 
-            // Add the new column to the table definition.
-            columns.push(new_column);
+            // Add the new column to the table definition if it is not created by `create table (*)` syntax.
+            if !columns.is_empty() {
```

Comment:
This doesn't look good to me. 😕

- I remember that we had a discussion long ago that we should _purify_ the definition of a table/source before persisting it into the catalog. In this case, for example, `*` is completely a syntax-sugar and should be first expanded to column definitions. cc @st1page 
- Considering empty columns as created with `*` is also a fragile assumption. Users are allowed to create a table without columns like `CREATE TABLE t;`.

Reply:
sorry for forgetting this point during my previous review of the https://github.com/risingwavelabs/risingwave/pull/16986. 
I think we should expand on `CREATE TABLE (*)`. This is because operations such as CREATE SINK INTO TABLE, ALTER TABLE RENAME, and others rely on reparsing the SQL statement, and the parsed columns should remain consistent with the previous plan.

Reply:
I think cdc table was the first to meet this problem, because in other scenarios with a schema registry, we simply prevented users from doing things like add columns.
https://github.com/risingwavelabs/risingwave/blob/main/src/frontend/src/handler/alter_table_column.rs#L117
https://github.com/risingwavelabs/risingwave/blob/main/src/frontend/src/handler/alter_source_column.rs#L73

And the only way to change their schema was to refresh the schema registry using alter table, which implied the assumption that their which implies the assumption that their schema needs to always be aligned upstream.

Reply:
Before finding a elegent way to support that. We can tentatively assume that the cdc table of create table(*) is similar to the table with connector of the schema registry. Let's temporarily prevent users from using add columns on such a table?

Reply:
I think for this PR we can prevent users from adding/drop columns on a table with wildcard.
And here is the future design https://github.com/risingwavelabs/risingwave/issues/17472

Reply:
> I think for this PR we can prevent users from adding/drop columns on a table with wildcard.

ok, we can ban `ALTER` for tables created by `create table(*)` in this pr.

Reply:
> ok, we can ban `ALTER` for tables created by `create table(*)` in this pr.

Does this also mean that we no longer need `let mut columns_altered = original_catalog.columns.clone();`?